### PR TITLE
CLOUDP-261715: Enabled resync for Atlas custom resources

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one). Closes https://github.com/mongodb/mongodb-atlas-kubernetes/issues/1689
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by s-urbaniak — CHANGES_REQUESTED** [review]
This is not enough:
1. Please add a resource version predicate on all Secret watches. Otherwise we will have double reconciliations in the face of resyncs.
2. Please make the resync interval configurable in the builder and add an e2e test.

**Comment by s-urbaniak** [comment]
re 2.: for now, we shouldn't expose the knob to main just yet but just for the e2e test.

**Comment by igor-karpukhin** [comment]
@s-urbaniak the sync interval is already configurable. The main just doesn't set it and it's set up to the default value of 3 hours

**Comment by igor-karpukhin** [comment]
> Otherwise we will have double reconciliations in the face of resyncs.
I just checked that locally, and it doesn't double the reconciliations. It doubles them if we actually remove the predicated

File: test/int/syncperiod_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+				previousResourceVersion = createdProject.ResourceVersion
+				return true
+			}).WithTimeout(10 * time.Second).WithPolling(2 * time.Second).Should(BeTrue())
+		})
+
+		By(fmt.Sprintf("Should wait for at least %f seconds", (syncInterval*2).Seconds()), func() {
+			time.Sleep(syncInterval * 2)
+		})
+
+		By("Project generation should be different", func() {
```
Comment (s-urbaniak) [inline test/int/syncperiod_test.go:89]:
```suggestion
		By("Project resource version should be different", func() {
```

**Review by s-urbaniak — COMMENTED** [review]
one nit, else lgtm 👏

File: test/int/syncperiod_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/conditions"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/events"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/resources"
+)
+
+var _ = Describe("Sync Period test", Label("int", "sync-period"), func() {
+	const interval = time.Second * 2
+	const syncInterval = 40 * time.Second
```
Comment (josvazg) [inline test/int/syncperiod_test.go:22]: nit pick
would 10s be enough to test the same?

**Review by josvazg — APPROVED** [review]

File: test/int/syncperiod_test.go (hunk 2 above)
Comment (s-urbaniak) [reply test/int/syncperiod_test.go:22]: As a side note: i am working on a pure unit-test approach for this. I don't think we need a full blown e2e test to assert resyncs. I will submit once ready.

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**